Update CityWithTruck.sql view creation script

odarmokhval opened the pull request:
Update CityWithTruck.sql view creation script

odarmokhval committed:
Update CityWithTruck.sql view creation script

odarmokhval committed:
Update file name to Task3

AnastasiaTitova-Int commented on Task3_ViewCreatingScript.sql:
```diff
@@ -0,0 +1,10 @@
+CREATE VIEW CityWithTruck AS
+SELECT
+    r.OriginWaterhouseId,
```
This view requires more information that you have
Also, in order to task 3 you should return not WarehouseIds, but city names

AnastasiaTitova-Int commented on Task3_ViewCreatingScript.sql:
```diff
@@ -0,0 +1,10 @@
+CREATE VIEW CityWithTruck AS
```
Make script reusable, similar to create DB script

AnastasiaTitova-Int commented:
Remove script Task1_VerifyingScript.sql it's not needed

AnastasiaTitova-Int commented:
Remove duplicated insert into Route table
![image](https://github.com/odarmokhval/hw1/assets/144356242/c157cde4-44a2-421f-ad7e-38381df057ef)

AnastasiaTitova-Int commented:
You can use PlaceState.csv file to populate those tables using temporal table. Try to use this approach
![image](https://github.com/odarmokhval/hw1/assets/144356242/8a1a2ca0-44a9-40c2-934b-c115d09a3107)

AnastasiaTitova-Int commented:
General question to Task2_PopulateDataBaseTables.sql
Are you sure that all your columns will be populated with proper data without additional casts to needed data types?

AnastasiaTitova-Int commented:
If you are setting random distance to all roads in this script, then you don't need previous logic of calculating road's distance
![image](https://github.com/odarmokhval/hw1/assets/144356242/ab6a6e99-5183-4171-a92d-d03bac8f207c)

AnastasiaTitova-Int commented:
For what purposes do you need PersonID in Place table? It shouldn't be here, I believe
![image](https://github.com/odarmokhval/hw1/assets/144356242/9ce5fca9-db76-479d-87cb-6d02a91e17e1)

odarmokhval committed:
1. Update view: returns Origin city, Destination city, Truck brand
2. Make script for creating view reusable
3. Remove script Task1_VerifyingScript.sql
4. Remove duplicated insert into Route table
5. Populating 'Place' and 'State' tables using temporary table
6. Change data types in tables: Place: PlaceCode -> char(2) to int, add PlaceName PlaceName  VARCHAR(255) NOT NULL,
Routes: Distance -> decimal to float, OriginWaterhouseId, DestinationWaterhouseId to int
7. Remove PersonID from the Task1_CreatingScript table
8. Extend table Warehouse: add CityName, StateName
9. Populating 'Route' using temporary table
10. Add CityName, StateName to Warehouse table
11. Add OriginalStateId,  DestinationStateId to Route table
